exonum/src/blockchain/mod.rs

```diff
@@ -183,12 +184,53 @@ impl Blockchain {
         let has_genesis_block = !Schema::new(&self.snapshot())
             .block_hashes_by_height()
             .is_empty();
-        if !has_genesis_block {
+        if has_genesis_block {
+            self.check_storage_version();
+        } else {
+            self.initialize_metadata();
             self.create_genesis_block(cfg)?;
         }
         Ok(())
     }
 
+    /// Initialized node-local metadata.
+    fn initialize_metadata(&mut self) {
+        let mut fork = self.db.fork();
+        let ver = storage::initialize_storage_version(&mut fork);
+        if self.merge(fork.into_patch()).is_ok() {
+            info!(
+                "Storage version successfully initialized with value [{}].",
+                ver
+            )
+        } else {
+            panic!("Could not set database version.")
+        }
+    }
+
+    /// Checks if storage version is supported.
+    ///
+    /// # Panics
+    ///
+    /// Panics if version is not supported or is not specified.
+    fn check_storage_version(&self) {
```

Comment:
This method should probably be named with `assert` prefix.

Reply:
`assert_*` is a good name. Can you please elaborate on the `maintenance` command? I can only see one usage of it - setting storage version when there is none set. But this scenario is very limited and this command will only be useful when migrating to 0.8.

Reply:
I think migration to 0.8 is a good case for this command.